examples/ars.js

```
@@ -0,0 +1,115 @@
+        $('*').contents().each(function () {
+            if(this.nodeType === COMMENT_NODE) {
+                $(this).remove();
+            }
+        });
+
+        let authors = $('header [itemprop~="author"] [itemprop~="name"]')
+            .map((i, elem) => $(elem).text())
+            .get();
+        asset.set_authors(authors);
+        $('header').remove();
+
+        let main_candidates = $('figure.intro-image');
+        if (!main_candidates.length)
+            main_candidates = $('figure');
+        if (main_candidates.length) {
+            let main = main_candidates.first();
+            let img = $('img', main);
+            let img_asset = Libingester.util.download_img(img, BASE_URI);
+            hatch.save_asset(img_asset);
+            asset.set_main_image(img_asset, $('figcaption', main).first());
+            $(main).remove();
+        }
+
+        let first_para = $('section p').first();
+        asset.set_lede(first_para);
+        $(first_para).remove();
+
+        // Clean up classes for readability
+        $('body, article, section, figure, figcaption').removeAttr('class');
+        $('article').removeAttr('itemscope');
+        $('article').removeAttr('itemtype');
+        $('figure').removeAttr('style');
+
+        // Save assets for any remaining figures
+        $('figure').each(function () {
+            let fig_asset = Libingester.util.download_img($('img', this), BASE_URI);
+            hatch.save_asset(fig_asset);
+        });
+
+        asset.set_body($('section'));
+
+        asset.render();
+        hatch.save_asset(asset);
+    });
+}
+
+function main() {
+    let hatch = new Libingester.Hatch('ars');
+    const FEED_URI = 'http://feeds.arstechnica.com/arstechnica/open-source';
+    RssToJson.load(FEED_URI)
+    .then(rss =>
+        Promise.all(rss.items.map(entry => ingest_article(hatch, entry))))
+    .then(() => hatch.finish())
+    .catch(err => console.log('there was an error', err));
```

You should re-throw the error or the caller can't tell we had a failure.